File: src/test/java/edu/born/overseer/repository/implementation/RegionRepositoryImplTest.java
package edu.born.overseer.repository.implementation;

import edu.born.overseer.repository.RegionRepository;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.dao.DataIntegrityViolationException;

import static edu.born.overseer.data.RegionTestData.*;
import static edu.born.overseer.data.TestDataUtil.INVALID_ID;
import static org.junit.jupiter.api.Assertions.*;

class RegionRepositoryImplTest extends AbstractRepositoryTest {

    @Autowired
    private RegionRepository regionRepository;

    @BeforeEach
    public void setUp() throws Exception {
        regionRepository.evictCache();
    }

    @Test
    void create() {
        var prepared = getPreparedCreate();
        var savedId = regionRepository.save(prepared).getId();
        prepared.setId(savedId);

        assertEquals(regionRepository.getById(savedId), prepared);
    }

    @Test
    void createDuplicate() {
        assertThrows(DataIntegrityViolationException.class, () -> regionRepository.save(getPreparedDuplicate()));
    }

    @Test
    void update() {
        var prepared = getPreparedUpdate();
        var updated = regionRepository.save(prepared);

        assertEquals(updated, prepared);
    }

    @Test
    void delete() {
        regionRepository.delete(REGION_1_ID);
        assertNull(regionRepository.getById(REGION_1_ID));
    }

    @Test
    void deleteNotExecute() {
        assertEquals(regionRepository.delete(INVALID_ID), Boolean.FALSE);
    }

    @Test
    void getById() {
        assertEquals(regionRepository.getById(REGION_1_ID), REGION_1);
    }

  /*  @Test
    void getAll() {
        assertThat(regionRepository.getAll(unlimitedPageLength()), contains(REGION_1,
                REGION_2,
                REGION_3,
                REGION_4,
                REGION_5,
                REGION_6,
                REGION_7,
                REGION_8,
                REGION_9,
                REGION_10,
                REGION_11,
                REGION_12,
                REGION_13));
    }
*/
}

File: src/test/java/edu/born/overseer/repository/impl/RegionRepositoryImplTest.kt
package edu.born.overseer.repository.impl

import edu.born.overseer.data.*
import edu.born.overseer.model.Region
import edu.born.overseer.repository.RegionRepository
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.contains
import org.junit.Test
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.dao.DataIntegrityViolationException
import java.lang.Boolean.FALSE

internal class RegionRepositoryImplTest : AbstractRepositoryTest() {

    @Autowired
    private lateinit var regionRepository: RegionRepository

    @BeforeEach
    fun setUp() {
        regionRepository.evictCache()
    }

    @Test
    fun create() {
        val prepared = getPreparedRegionCreate()
        val savedId = regionRepository.save(prepared).id
        prepared.id = savedId

        assertEquals(regionRepository.getById(savedId), prepared)
    }

    @Test
    fun createDuplicate() {
        val duplicate = getPreparedRegionCreate().apply {
            title = "Moscow"
        }

        assertThrows(DataIntegrityViolationException::class.java) {
            regionRepository.save(duplicate)
        }
    }

    @Test
    fun update() {
        val prepared = Region(REGION_1).apply {
            title = "Updated title"
        }
        val updated = regionRepository.save(prepared)

        assertEquals(updated, prepared)
    }

    @Test
    fun delete() {
        regionRepository.delete(REGION_1_ID)

        assertNull(regionRepository.getById(REGION_1_ID))
    }

    @Test
    fun deleteNotExecute() {
        assertEquals(regionRepository.delete(INVALID_ID), FALSE)
    }

    @Test
    fun getById() {
        assertEquals(regionRepository.getById(REGION_1_ID), REGION_1)
    }

    @Test
    fun getAll() {
        assertThat(regionRepository.getAll(1, "Mo"), contains(REGION_2))
    }
}